packages/core/src/types.ts

```diff
@@ -125,297 +35,177 @@ export interface IKeyframesAtom {
-  colors?: ITokenDefinition;
-  space?: ITokenDefinition;
-  fontSizes?: ITokenDefinition;
-  fonts?: ITokenDefinition;
-  fontWeights?: ITokenDefinition;
-  lineHeights?: ITokenDefinition;
-  letterSpacings?: ITokenDefinition;
-  sizes?: ITokenDefinition;
-  borderWidths?: ITokenDefinition;
-  borderStyles?: ITokenDefinition;
-  radii?: ITokenDefinition;
-  shadows?: ITokenDefinition;
-  zIndices?: ITokenDefinition;
-  transitions?: ITokenDefinition;
+export type CSSPropertyKeys = keyof Properties;
+export interface CSSPropertiesToTokenScale {
+  gap: 'space';
+  gridGap: 'space';
+  columnGap: 'space';
+  gridColumnGap: 'space';
+  rowGap: 'space';
+  gridRowGap: 'space';
+  inset: 'colors';
+  insetBlock: 'colors';
+  insetBlockEnd: 'colors';
+  insetBlockStart: 'colors';
+  insetInline: 'colors';
+  insetInlineEnd: 'colors';
+  insetInlineStart: 'colors';
+  margin: 'space';
+  marginTop: 'space';
+  marginRight: 'space';
+  marginBottom: 'space';
+  marginLeft: 'space';
+  marginInline: 'space';
+  marginInlineEnd: 'space';
+  marginInlineStart: 'space';
+  marginBlock: 'space';
+  marginBlockEnd: 'space';
+  marginBlockStart: 'space';
+  padding: 'space';
+  paddingTop: 'space';
+  paddingRight: 'space';
+  paddingBottom: 'space';
+  paddingLeft: 'space';
+  paddingInline: 'space';
+  paddingInlineEnd: 'space';
+  paddingInlineStart: 'space';
+  paddingBlock: 'space';
+  paddingBlockEnd: 'space';
+  paddingBlockStart: 'space';
+  top: 'space';
+  right: 'space';
+  bottom: 'space';
+  left: 'space';
+  fontSize: 'fontSizes';
+  backgroundColor: 'colors';
+  /** Update syntax when ts 4.1 is release */
+  // border: ''
+  borderColor: 'colors';
+  borderTopColor: 'colors';
+  borderRightColor: 'colors';
+  borderBottomColor: 'colors';
+  borderLeftColor: 'colors';
+  caretColor: 'colors';
+  color: 'colors';
+  columnRuleColor: 'colors';
+  outlineColor: 'colors';
+  fill: 'colors';
+  stroke: 'colors';
+  fontFamily: 'fonts';
+  fontWeight: 'fontWeights';
+  lineHeight: 'lineHeights';
+  letterSpacing: 'letterSpacings';
+  blockSize: 'sizes';
+  minBlockSize: 'sizes';
+  maxBlockSize: 'sizes';
+  inlineSize: 'sizes';
+  minInlineSize: 'sizes';
+  maxInlineSize: 'sizes';
+  width: 'sizes';
+  minWidth: 'sizes';
+  maxWidth: 'sizes';
+  height: 'sizes';
+  minHeight: 'sizes';
+  maxHeight: 'sizes';
+  flexBasis: 'sizes';
+  /** Update syntax when ts 4.1 is release */
+  // flex:'';
+  borderWidth: 'borderWidths';
+  borderTopWidth: 'borderWidths';
+  borderLeftWidth: 'borderWidths';
+  borderRightWidth: 'borderWidths';
+  borderBottomWidth: 'borderWidths';
+  borderStyle: 'borderStyles';
+  borderTopStyle: 'borderStyles';
+  borderRightStyle: 'borderStyles';
+  borderBottomStyle: 'borderStyles';
+  borderLeftStyle: 'borderStyles';
+  borderRadius: 'radii';
+  borderTopLeftRadius: 'radii';
+  borderTopRightRadius: 'radii';
+  borderBottomRightRadius: 'radii';
+  borderBottomLeftRadius: 'radii';
+  /** Update syntax when ts 4.1 is release */
```

Comment:
what to does comment mean?

Also, typo nit: `released`

Reply:
I believe typesciprt 4.1 RC is out now so we should be able to leverage on https://devblogs.microsoft.com/typescript/announcing-typescript-4-1-beta/#template-literal-types
to improve the types for similar properties